src/console_cmds.cpp

```diff
@@ -53,6 +53,20 @@
 /* scriptfile handling */
 static uint _script_current_depth; ///< Depth of scripts running (used to abort execution when #ConReturn is encountered).
 
+/* Scheduled execution handling */
+static std::string _scheduled_filename;
+static IntervalTimer<TimerGameCalendar> _schedule_timer = {{TimerGameCalendar::MONTH, TimerGameCalendar::Priority::NONE}, [](auto) {
+	if (_scheduled_filename != "") {
+		/* Clear the schedule before rather than after the script to allow the script to itself call
+		 * schedule without it getting immediately cleared */
+		const std::string filename = _scheduled_filename;
+		_scheduled_filename = "";
+
+		IConsolePrint(CC_DEFAULT, "Executing scheduled script file '{}'...", filename);
+		IConsoleCmdExec(std::string("exec") + " " + filename);
+	}
```

Comment:
Use `empty()` and `clear()` instead of `""`.
Prefer early return instead of nesting, i.e. return if the script is not set.